project setup

leonard33 opened the pull request:
setting up project

leonard33 committed:
project setup

leonard33 committed:
add linters file

leonard33 committed:
fix linters

leonard33 committed:
fix linters

leonard33 committed:
fix linters

leonard33 committed:
fix linters

leonard33 committed:
fix linters

leonard33 committed:
fix linters

leonard33 committed:
add components and styling

leonard33 committed:
fix linters

leonard33 committed:
fix linters

phelian23 commented on src/Components/Nav.js:
```diff
@@ -0,0 +1,33 @@
+          <li>
+            <a href="#">CATEGORIES</a>
+          </li>
+        </ul>
```
- [ ] The links should be leading to the booklist and category locations. Kindly make sure that the links when clicked, are linked to each page. You'll need to use `BrowserRouter, Routes, and Route` from `react-router-dom`.

phelian23 commented on src/Components/Booklist.js:
```diff
@@ -0,0 +1,29 @@
+        booktitle="Capital in the Twenty First Century"
+        author="Frank Herbert"
+        chapter="Introduction"
+      />
```
- [ ] Instead of having to call the `Bookitem` components every time you want to render a book, it will be better if you have an array for the list of books and iterate through the list of books to get the details of each book. eg.
``` javascript
  const books = [
    {
       id: 1,
       title: "Something",
       author: "Something",
    },
   {
       id: 1,
       title: "Something",
       author: "Something",
    }
  ]
// then you can have something like
  <ul>
       { books.map((book) => (<Bookitem key={book.id} title={book.title} author={book.author} />))}
  </ul>
```
That way you won't have to repeat yourself.

phelian23 commented on src/Components/Form.js:
```diff
@@ -0,0 +1,19 @@
+        <input type="text" placeholder="AUTHOR" />
+        <a className="btn-submit" input type="button">
+          ADD BOOK
+        </a>
```
- [ ] Kindly use a `<button>` element or an `<input>` element with a type submit for the submit button.

phelian23 commented on src/Components/Bookitem.js:
```diff
@@ -0,0 +1,42 @@
+        <div className="percentage">
+          <div className="number">{percentage}</div>
+          <div className="percentage-detail">completed</div>
+  </div>*/}
```
- [ ] Kindly remove commented block of code as it'll make your code cleaner

phelian23 commented on src/Components/Bookitem.js:
```diff
@@ -0,0 +1,42 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+import "./Bookitem.css";
+// import { AiOutlineLoading3Quarters } from "react-icons/ai";
```
- [ ] Kindly remove commented block of code as it'll make your code cleaner

phelian23 reviewed (changes requested):
Hi @leonard33, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

leonard33 committed:
correction as per review

leonard33 committed:
fix linters

leonard33 commented:
corrections PR:
- Added react router and links
-Deleted commented code
- looped and iterated through the books for rendering
- changed form button to a button element as per review

phelian23 reviewed (changes requested):
Hi @leonard33, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

leonard33 commented:
I have done all the changes you requested . Thats all I think?


   -  Kindly remove commented block of code as it'll make your code cleaner
   -  Kindly remove commented block of code as it'll make your code cleaner
   -  Kindly use a <button> element or an <input> element with a type
   submit for the submit button.
   -

     <ul>

          { books.map((book) => (<Bookitem key={book.id}
title={book.title} author={book.author} />))}

     </ul>

   That way you won't have to repeat yourself.
   -

    The links should be leading to the booklist and category locations.
   Kindly make sure that the links when clicked, are linked to each page.
   You'll need to use BrowserRouter, Routes, and Route from react-router-dom
   .


On Fri, Jun 3, 2022 at 6:00 PM Oluwafemi Awoyemi ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi @leonard33 <https://github.com/leonard33>,
>
> Great work on making the changes requested by a previous reviewer 👏
> You've done well implementing some of the requested changes, but there are
> still some which aren't addressed yet.
> Required Changes ♻️
>
> *Check the comments under the review.*
> Optional suggestions
>
> *Every comment with the [OPTIONAL] prefix won't stop the approval of this
> PR. However, I strongly recommend you consider them as they can make your
> code better. Some of them were simply missed by the previous reviewer and
> addressing them will really improve your application.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
> Please, remember to tag me in your question so I can receive the
> notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/leonard33/Bookstore-cms/pull/1#pullrequestreview-995034678>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD6Z6L4WOVRS46DTL5F5QQTVNIMZZANCNFSM5XXF4RUA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

*RegardsLEONARD .M. KOMBO*

phelian23 commented:
@leonard33 yes you did all but one... The first comment stated that the link to the booklist and categories should be leading to their respective pages. You did that of the booklist, but yet to do that of the categories...Kindly add it.
Thanks

leonard33 committed:
review changes

leonard33 committed:
review changes

leonard33 commented:
- link to categories added

phelian23 commented:
> Hi @leonard33,
> 
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Oh sorry, I added an inline comment to this to show where the change is required, don't know it's not showing somehow...just noticed that.

leonard33 committed:
add read me file

RolandM99 reviewed (approved):
<h1>Status: Approve 🥇 ✔</h1>
Hi @leonard33 , 

### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

<img src="https://media.giphy.com/media/aNqEFrYVnsS52/giphy.gif" width="100%">

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

leonard33 commented:
Hallo I thought as per the directions this was just supposed to be a setup
to initialise the components as we ait for the next step?????????

On Fri, Jun 3, 2022 at 4:23 PM Oluwafemi Awoyemi ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi @leonard33 <https://github.com/leonard33>,
>
> Good job so far!
> You still need to work on some issues to go to the next project, but you
> are almost there!
> Required Changes ♻️
>
> *Check the comments under the review.*
> Optional suggestions
>
> *Every comment with the [OPTIONAL] prefix is not crucial enough to stop
> the approval of this PR. However, I strongly recommend you consider them as
> they can make your code better.*
>
> You can also consider:
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
> Please, remember to tag me in your question so I can receive the
> notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
> ------------------------------
>
> In src/Components/Nav.js
> <https://github.com/leonard33/Bookstore-cms/pull/1#discussion_r888920797>:
>
> > +        <ul className="menu-list">
>
> +          <li>
>
> +            <a href="#">BOOKS</a>
>
> +          </li>
>
> +          <li>
>
> +            <a href="#">CATEGORIES</a>
>
> +          </li>
>
> +        </ul>
>
>
>
>    - The links should be leading to the booklist and category locations.
>    Kindly make sure that the links when clicked, are linked to each page.
>    You'll need to use BrowserRouter, Routes, and Route from
>    react-router-dom.
>
> ------------------------------
>
> In src/Components/Booklist.js
> <https://github.com/leonard33/Bookstore-cms/pull/1#discussion_r888929706>:
>
> > +      <Bookitem
>
> +        genre="Action"
>
> +        booktitle="The Hunger Games"
>
> +        author="Suzzane Collins"
>
> +        chapter="Chapter 17"
>
> +      />
>
> +      <Bookitem
>
> +        genre="Science Fiction"
>
> +        booktitle="Dune"
>
> +        author="Frank Herbert"
>
> +        chapter="Chapter 3: Lesson learned"
>
> +      />
>
> +      <Bookitem
>
> +        genre="Science Fiction"
>
> +        booktitle="Capital in the Twenty First Century"
>
> +        author="Frank Herbert"
>
> +        chapter="Introduction"
>
> +      />
>
>
>
>    - Instead of having to call the Bookitem components every time you
>    want to render a book, it will be better if you have an array for the list
>    of books and iterate through the list of books to get the details of each
>    book. eg.
>
>   const books = [
>
>     {
>
>        id: 1,
>
>        title: "Something",
>
>        author: "Something",
>
>     },
>
>    {
>
>        id: 1,
>
>        title: "Something",
>
>        author: "Something",
>
>     }
>
>   ]
> // then you can have something like
>
>   <ul>
>
>        { books.map((book) => (<Bookitem key={book.id} title={book.title} author={book.author} />))}
>
>   </ul>
>
> That way you won't have to repeat yourself.
> ------------------------------
>
> In src/Components/Form.js
> <https://github.com/leonard33/Bookstore-cms/pull/1#discussion_r888934959>:
>
> > +        <a className="btn-submit" input type="button">
>
> +          ADD BOOK
>
> +        </a>
>
>
>
>    - Kindly use a <button> element or an <input> element with a type
>    submit for the submit button.
>
> ------------------------------
>
> In src/Components/Bookitem.js
> <https://github.com/leonard33/Bookstore-cms/pull/1#discussion_r888935917>:
>
> > +        {/*<div className="comlete-icon">
>
> +          <AiOutlineLoading3Quarters size={70} color={"blue"} />
>
> +        </div>
>
> +        <div className="percentage">
>
> +          <div className="number">{percentage}</div>
>
> +          <div className="percentage-detail">completed</div>
>
> +  </div>*/}
>
>
>
>    - Kindly remove commented block of code as it'll make your code cleaner
>
> ------------------------------
>
> In src/Components/Bookitem.js
> <https://github.com/leonard33/Bookstore-cms/pull/1#discussion_r888936747>:
>
> > @@ -0,0 +1,42 @@
>
> +/* eslint-disable react/prop-types */
>
> +import React from "react";
>
> +import "./Bookitem.css";
>
> +// import { AiOutlineLoading3Quarters } from "react-icons/ai";
>
>
>
>    - Kindly remove commented block of code as it'll make your code cleaner
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/leonard33/Bookstore-cms/pull/1#pullrequestreview-994880713>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD6Z6L6VJI5DMNRNAL7LAHLVNIBLHANCNFSM5XXF4RUA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

*RegardsLEONARD .M. KOMBO*
